src/librustc/middle/ty/util.rs

```diff
@@ -566,84 +566,25 @@ impl<'tcx> ty::ctxt<'tcx> {
         }
     }
 
-    /// Returns true if this ADT is a dtorck type, i.e. whether it being
-    /// safe for destruction requires it to be alive
+    /// Returns true if this ADT is a dtorck type, i.e. whether it
+    /// being safe for destruction requires all borrowed pointers
+    /// reachable by it to have lifetimes strictly greater than self.
```

Comment:
nit: I used "alive" intentionally - phantom lifetimes are just as important as borrowed pointers here. "Greater than self" is also not entirely accurate - you can *totally* have a `&'a mut MyTrait+'a` (or even a `&'a mut TypeWithDestructor<'a>`, if you are willing to use unsafe means to actually *create* and *destroy* it), you just can't call its destructor outside `'a`.

Maybe 
```Rust
    /// Returns whether this ADT is a dtorck type. If not, it
    /// is safe to call this type's destructor even when it does
    /// not actually outlive the call, as long as its contents
    /// remain destruction-safe.
```